lib/pacemaker/pcmk_sched_location.c

```diff
@@ -196,15 +196,6 @@ generate_location_rule(pcmk_resource_t *rsc, xmlNode *rule_xml,
         }
     }
 
-    if (combine == pcmk__combine_and) {
```

Comment:
That is just odd. I'm wary of this -- I want to think there's some reason for the existing design -- but like you, I don't see any difference between this and switching to the "or" logic.

I suspect this is just an artifact of confusion from how much more complicated this code used to be.